### PR TITLE
Add Collector API datastore

### LINKED ISSUE #1109: Feature: Datastore integration with `Collector API`
An external Collector API tool (https://github.com/Karm/collector) allows performance metrics to be pushed and queried as JSON documents via a REST API

In order to integrate existing datasets into Horreum for analysis, a new `Collector API` datastore needs to be created that allows Horreum Users to query and existing instance.

The `Collector API` datastore should be defined as a new Datastore Type (https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/datastore/Datastore.java), similar to how the Elasticsearch datastore is integrated (https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/datastore/ElasticsearchDatastore.java)

Documentation on how the Elasticsearch document store is used can be found here: https://horreum.hyperfoil.io/docs/tutorials/query-elasticserach/

### PR DESCRIPTION
**Pull request by zakkak**
## Fixes Issue

Closes #1109 

## Proposed Changes

This patch issues a request to the collector API to get all available data. To limit the size of returned data we allow the use of a specific `tag` and `imgName` combination, while also limiting the date range.

## Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [X] My change requires changes to the documentation.
- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

File: horreum-backend/src/main/java/io/hyperfoil/tools/horreum/datastore/CollectorApiDatastore.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            }
+            metaData = payload;
+
+            final CollectorApiDatastoreConfig jsonDatastoreConfig = mapper.treeToValue(configuration.configuration, CollectorApiDatastoreConfig.class);
+
+            if ( jsonDatastoreConfig != null ){
+
+                HttpClient client = HttpClient.newHttpClient();
+                HttpRequest.Builder builder = HttpRequest.newBuilder()
+                        .uri(java.net.URI.create(jsonDatastoreConfig.url + "/api/v1/image-stats/tag/" + payload.get("tag").asText()));
```
Comment (johnaohara) [inline horreum-backend/src/main/java/io/hyperfoil/tools/horreum/datastore/CollectorApiDatastore.java:51]: should the url `path` be a separate config param, or part of the `jsonDatastoreConfig.url`, if the api changes (e.g. a version bump) then this hard-coded path will likely be no longer valid

**Review by johnaohara — CHANGES_REQUESTED** [review]
One thing we need to think about is duplicated runs. As it currently works, the call to the collector api simply returns all the results for a given `tag`, each invocation will return all the data and will be uploaded into Horreum

So, we either need to be able to query the api to return a subset of the data, or we need to handle de-duplication in Horreum

**Comment by johnaohara** [comment]
Ah, I now see this issue wrt filtering the api result: https://github.com/Karm/collector/pull/34

**Comment by zakkak** [comment]
https://github.com/Karm/collector/pull/35 is now merged enabling filtering by tag, benchmark, and date range (not pushed to production yet though). So I will adapt the PR to use the new API.

**Comment by johnaohara** [comment]
@zakkak excellent news.

**Comment by zakkak** [comment]
@johnaohara this should now be ready for review.

**Review by johnaohara — APPROVED** [review]
LGTM!

**Comment by johnaohara** [comment]
@zakkak Many thanks for your contribution!

**Pull request merged**